src/bogart/AS_BAT_BestOverlapGraph.C

```diff
@@ -134,77 +214,92 @@ BestOverlapGraph::findErrorRateThreshold(void) {
   //  Find the median and absolute deviations.
 
   computeMedianAbsoluteDeviation(erates, _median, _mad, true);
-  uint32   pos = (uint32)((erates.size()+1)*0.90);
 
-  //  Compute an error limit based on the median or absolute deviation or quantile
-  //    Use quantile is the median isn't zero.
-  //    Use mean+stddev otherwise.
+  //  Pick an error threshold.
+  //
+  //  The tight threshold is the minimum of:
+  //    the user supplied threshold (-eg or -eM as decided above)
+  //    the median + mad (if the median isn't zero)
+  //    the 90th percentile error rate (if the median is zero)
   //
-  //    But either way, let the user lower this via the command line.
+  //  The loose threshold is the minimum of:
+  //    the user supplied threshold (-eg or -eM as decided above)
+  //    the mean + stddev
+  //
+  //  Note that there is no guarantee that TpickedTight <= TpickedLoose!
 
-  double   Tinput  = _errorLimit;
+  double   Tgraph  = _erateGraph;   //  was Tinput = _errorLimit
+  double   Tmax    = _erateMax;
   double   Tmean   = _mean   + _deviationGraph          * _stddev;
   double   Tmad    = _median + _deviationGraph * 1.4826 * _mad;
+  uint32    pos    = (uint32)((erates.size()+1)*0.90);
   double   Tperct  = erates[pos] + 1e-5;
-  double   Tpicked = (_median > 1e-10) ? Tmad : Tperct;
 
-  if (Tinput < Tpicked)
-    _errorLimit = Tinput;
-  else
-    _errorLimit = Tpicked;
+  assert((_erateGraph == _errorLimit) ||   //  Either erateGraph or erateMax, as
+         (_erateMax   == _errorLimit));    //  set in findInitialEdges().
 
-  //  The real filtering is done on the next pass through findEdges().  Here, we're just collecting statistics.
-  writeLog("\n");
-  writeLog("ERROR RATES\n");
-  writeLog("-----------\n");
-  writeLog("\n");
+  double   TpickedTight = min(_errorLimit, ((_median > 1e-10) ? Tmad : Tperct));
+  double   TpickedLoose = min(_errorLimit, Tmean);
+  double   Tpicked      = 0.0;
 
-  uint32  nFiltered[3] = {0};
-  uint32  nTotal = 0;
-  bool    adjustedLimit = false;
+  //  Summarize the number of reads with edges.  If there are too few reads
+  //  that have both ends with valid edges, increase the picked error
+  //  threshold to the mean.
 
-  while (nTotal == 0 && adjustedLimit == false) {
-    for (uint32 fi=1; fi <= fiLimit; fi++) {
-      BestEdgeOverlap *b5 = getBestEdgeOverlap(fi, false);
-      BestEdgeOverlap *b3 = getBestEdgeOverlap(fi, true);
-
-      if ((b5->isUnset() == true) &&    //  Read has no best edges, probably
-          (b3->isUnset() == true))      //  not corrected.
-        continue;
+  uint32  nFilteredTight[4] = {0};
+  uint32  nFilteredLoose[4] = {0};
 
-      uint32  nf = ((b5->erate() > _errorLimit) +
-                    (b3->erate() > _errorLimit));
+  bool  tightGood = summarizeBestEdges(TpickedTight, 0.90, nFilteredTight);   //  Good if more than 90% of the
```

Comment:
Since this code is more consolidated, can we make the 0.9 repeated three times a #define or maybe a parameter?

Reply:
And make it 0.8 instead of 0.9 used currently

Reply:
Done.  It's a bogart parameter, but not exposed directly to canu.